app/services/ratings.js

```diff
@@ -0,0 +1,47 @@
+const { User, Weet, Rating } = require('../models');
+const { sequelize } = require('../models');
+const logger = require('../logger');
+
+exports.findRateWeet = async (id, ratingUserId) => {
+  const rating = await Rating.findOne({ where: { weetId: id, ratingUserId } });
+  return rating;
+};
+
+exports.createRateWeet = async (previousRateId, weetId, body) => {
+  let transaction = {};
+  const { score, ratingUserId } = body;
+  try {
+    transaction = await sequelize.transaction();
+    const currentRatedWeet = await Rating.upsert(
+      {
+        id: previousRateId,
+        ratingUserId,
+        score,
+        weetId
+      },
+      {
+        where: { id: previousRateId },
+        returning: true
+      }
+    );
+    const weet = await Weet.findOne(
+      {
+        include: [{ model: User, required: true, attributes: ['id', 'points'] }],
+        returning: true
+      },
+      { where: { id: weetId } }
+    );
+    logger.info(`actual weet is: ${weet}`);
+    const weetUserId = weet && weet.User.id;
+    const currentPoints = weet && weet.User.points;
+    await User.update(
+      { points: currentPoints + score },
```

Comment:
If the rate change, are you replacing the previous rate or just adding it?

Reply:
if the rate changes, the user's points are updated and the previous rate too.